prow/cmd/webhook-server/clients.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"k8s.io/test-infra/experiment/clustersecretbackup/secretmanager"
+)
+
+type GCPClient struct{
+	client *secretmanager.Client
+}
+
+func NewGCPClient(client *secretmanager.Client) *GCPClient {
+	return &GCPClient{
+		client: client,
+	}
+}
+
+func (g *GCPClient) CreateSecret(ctx context.Context, secretID string) error {
+	_, err := g.client.CreateSecret(ctx, secretID)
+	if err != nil {
+		return fmt.Errorf("could not create secret %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) AddSecretVersion(ctx context.Context, secretName string, payload []byte) error {
+	if err := g.client.AddSecretVersion(ctx, secretName, payload); err != nil {
+		return fmt.Errorf("could not add secret data %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) GetSecretValue(ctx context.Context, secretName string, versionName string) ([]byte, error) {
+	res, err := g.client.GetSecretValue(ctx, secretName, versionName)
+	if err != nil {
+		var found bool
+		res, err := g.client.ListSecrets(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("could not make call to list secrets successfully %v", err)
+		}
+		for _, secret := range(res) {
+			if secretID == secret.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return nil, fmt.Errorf("secret %s was not created %v", secretID, err)
+		}
+	}
+	return res, nil
+}
+
+type localFSClient struct {
+	path   string
+	expiry int
+	dns    []string
+}
+func NewLocalFSClient(path string, expiry int, dns []string) *localFSClient {
+	return &localFSClient{
+		path: path,
+		expiry: expiry,
+		dns: dns,
+	}
+}
+
+func (l *localFSClient) CreateSecret(ctx context.Context, secretID string) (error) {
+	if _, err := os.Stat(l.path); errors.Is(err, os.ErrNotExist) {
```

Comment:
nit: it would be more natural to allow multiple secret in this localFSClient if it's desired to mimic the GCP secret manager. So `localFSClient.path` is the root path for all the secrets, and `newLocalFSClient()` is responsible for ensuring that this directory is created. And `CreateSecret()` function creates directory named `secretID` underneath the root dir, it would fail if this dir already exists.

understand that this is for integration test purpose, so it might not be critical, up to you to decide.